tests/graphicsItems/test_PlotDataItem.py

```diff
@@ -71,6 +72,35 @@ def test_setData():
     pdi.setData([],[])
     assert pdi.xData is None
     assert pdi.yData is None
+    
+def test_nonfinite():
+    def _assert_equal_arrays(a1, a2):
+        assert a1.shape == a2.shape
+        for ( xtest, xgood ) in zip( a1, a2 ):
+            assert( (xtest == xgood) or (np.isnan(xtest) and np.isnan(xgood) ) ) 
+        
+    x = np.array([-np.inf, 0.0, 1.0,  2.0  , np.nan,   4.0 , np.inf])
+    y = np.array([    1.0, 0.0,-1.0, np.inf,   2.0 , np.nan,   0.0 ])
```

Comment:
This should look for crashes with combinations of infs, NaNs and regular values in normal and log mode.